### ISSUE TITLE
Deleted Contacts

### ISSUE THREAD

**Issue by Schmiddiii** [whisperfish/presage#214]
Signal now supports the deletion of contacts since a few versions (see https://github.com/signalapp/Signal-Desktop/releases/tag/v6.37.0). Currently, it seems like deleted contacts are still synced to presage and can be queried from the store.

Deleted contacts should not be stored after a contact sync and therefore also never be returned from the store using the corresponding methods.

**Comment by gferon** [comment]
To summarize, would this mean that you delete a contact and then mark it as "blacklisted" in the way that you will never save it again?

**Comment by Schmiddiii** [comment]
Yes, store some kind of tombstone to mark the contact as deleted and not sync it again.

I am surprised though that Signal even syncs such deleted contacts in full and does not send a tombstone for them. This may require looking into how Signal marks such contacts as deleted.
